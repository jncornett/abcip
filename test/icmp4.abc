# net unreachable
d(stack="eth:ip4:icmp4:ip4:udp")
a(icmp4:type=3, code=0, u32=0x12345678; len=16)

# ping
d(stack="eth:ip4:icmp4")
a(fill=64)
