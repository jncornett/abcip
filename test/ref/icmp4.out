IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto ICMP (1), length 72)
    10.1.2.3 > 10.9.8.7: ICMP net 10.9.8.7 unreachable, length 52
	IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto UDP (17), length 44)
    10.1.2.5.48620 > 10.9.8.7.8: UDP, length 16
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  0048 0001 0000 4001 5ca1 0a01 0203 0a09  .H....@.\.......
	0x0020:  0807 0300 b292 1234 5678 4500 002c 0001  .......4VxE..,..
	0x0030:  0000 4011 5cab 0a01 0205 0a09 0807 bdec  ..@.\...........
	0x0040:  0008 0018 dcd9 4c6f 7265 6d20 6970 7375  ......Lorem.ipsu
	0x0050:  6d20 646f 6c6f                           m.dolo
IP (tos 0x0, ttl 64, id 1, offset 0, flags [none], proto ICMP (1), length 92)
    10.1.2.3 > 10.9.8.7: ICMP echo request, id 1, seq 1, length 72
	0x0000:  0209 0807 0605 0201 0203 0405 0800 4500  ..............E.
	0x0010:  005c 0001 0000 4001 5c8d 0a01 0203 0a09  .\....@.\.......
	0x0020:  0807 0800 13fa 0001 0001 0001 0203 0405  ................
	0x0030:  0607 0809 0a0b 0c0d 0e0f 1011 1213 1415  ................
	0x0040:  1617 1819 1a1b 1c1d 1e1f 2021 2223 2425  ...........!"#$%
	0x0050:  2627 2829 2a2b 2c2d 2e2f 3031 3233 3435  &'()*+,-./012345
	0x0060:  3637 3839 3a3b 3c3d 3e3f                 6789:;<=>?
